bin/xbps-repodb/main.c

```diff
@@ -0,0 +1,122 @@
+/*-
+ * Copyright (c) 2020 Duncan Overbruck <mail@duncano.de>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE AUTHOR ``AS IS'' AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
+ * NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+ * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <strings.h>
+#include <errno.h>
+#include <unistd.h>
+#include <getopt.h>
+
+#include <xbps.h>
+
+#include "defs.h"
+
+
+static void __attribute__((noreturn))
+usage(void)
+{
+	fprintf(stdout,
+	"Usage: xbps-repodb [OPTIONS] MODE <repository>...\n\n"
+	"OPTIONS:\n"
+	" -d, --debug    Enable debug messages to stderr\n"
+	" -n, --dry-run  Dry-run mode\n"
+	" -v, --verbose  Enable verbose output\n"
+	" -V, --version  Prints the xbps release version\n"
+	"MODE:\n"
+	" -p, --purge    Remove obsolete binary packages from repositories\n");
+	exit(EXIT_FAILURE);
+}
+
+int
+main(int argc, char **argv)
+{
+	int c = 0, rv = 0;
+	struct xbps_handle xh;
```

Comment:
Can't you do `struct xbps_handle xh = { };` instead of a memset below? It should initialize the whole struct to zero and is easier to follow.

Reply:
Well, this isn't ISO C. `struct xbps_handle xh = { 0 };` should work too.